### DISCUSSION TITLE
Grid and AC loads meters doesn't show correct information after installing this driver

### DISCUSSION THREAD

**Discussion by Manuel-Calero** [mr-manuel/venus-os_dbus-enphase-envoy#9] [Q&A] [answered]
Hello,

First of all, thank you to mr-manuel for making this driver public.

I have a trifasic ESS system using three Multiplus II 5000-70 with 30 enphase microinverters (AC coupled).
I can install and uninstall without any problem your driver into my Cerbo GX and the PV inverter appears within seconds showing correct information.

However, once PV inverter appears, the information for GRID and AC loads is totally wrong.

It can be shown at the attached images.
![IMG_7505](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/971f9023-b600-435c-9b26-78d0c47a915a)
![IMG_7508](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/630e2357-ec8e-4fe5-bd3f-b48b9dffaf68)

In the first image you can see enlighten showing that microinverters are generating 2,4Kw and we are not using the grid. In the second, VRM shows PV inverter with 2444 (the same as enlighten shows) and the batteries charging at 800W. These means that AC Loads would be 2444-800=1644 (this is the real AC loads in my house) while VRM information for AC Loads shows 0W. Grid is showing -2389 W and must show 0W (the same as enlighten).

If we get the information from the inverters using remote console we have the correct information as you can see:
![IMG_7507](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/11957546-6124-4461-9a4a-16dc9ad8b291)
![IMG_7506](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/4af929ef-86e3-46c5-bd83-7809cf6fa190)

In the first image, grid power is 61w (almost 0) and in the second one, AC Loads is -834W because this is the part of PV generation that is not feeding AC loads and is used to charge the battery.

I have uninstalled driver and the information is correct again (but I cannot see the PV inverter):

![IMG_7513](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/979eba68-23b1-429c-8df2-975a2d7a3fef)
![IMG_7514](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/68d96269-6649-4bcb-8cb6-9263797adf5b)

Now PV is generating 3,6kw, using 1,8 to feed AC Loads and the other 1,8kw to charge the battery. Note that Victron AC Loads are not showing the kw that PV is inyecting because inverters are not inverting, they are charging.

Any idea for solving this situation?

Thank you!

Manuel Calero from Spain

**Comment by Manuel-Calero** [comment]
Only an update to my last paragraph. Victron AC Loads are showing 0 kw BUT, in the remote console, the information is right (I think this is a Victron issue). For example, as you can see in these three pictures:

![IMG_7515](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/2e3a7422-efc6-4232-918a-ed3046c42a27)
![IMG_7516](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/643b9495-6561-489e-9646-b7a19f433176)
![IMG_7517](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/262c2248-2743-4ff3-8b7d-f33da9038bbf)

My enphase system is generating 4,7kw. Victron VRM shows it is charging at 1398 w, so the real loads in my house are the result of 4,7-1,4=3,3kw. However, the AC load from the point of view of Victron is -1398 (the excess of PV system who is AC coupled with the loads in the house), and it is shown in the console view (-1408). So, the conclusion is that Victron Multiplus are measuring well (-1408) but this information is not shown correctly at the AC loads information. When AC loads are positive, the information is correctly shown at VRM.

Hope this help.

Thank you

**Comment by mr-manuel** [comment]
Please post your `config.ini`and also a technical schema of how your installation is cabeled.

**Comment by Manuel-Calero** [comment]
Hello. Here you are my config.ini file (login, password and last digits of S/N have been replaced by X) and a scheme showing how my installation is cabled and the position of enphase CT.

One more thing. In this link https://community.victronenergy.com/questions/63661/victron-portal-not-displaying-ac-load-and-power-co.html, you can check that despite configuring you have an AC coupled FV in your Victron you also have to declare the FV. However, Victron doesn't detect enphase system and I think this is the reason why it doesn't measure in the AC LOADS box when FV is injecting (and it shows zero).

<details><summary>config.txt</summary>

```ini
; CONFIG FILE
; GitHub reporitory: https://github.com/mr-manuel/venus-os_dbus-enphase-envoy
; remove semicolon ; to enable desired setting

[DEFAULT]
; Set logging level
; ERROR = shows errors only
; WARNING = shows ERROR and warnings
; INFO = shows WARNING and running functions
; DEBUG = shows INFO and data/values
; default: WARNING
logging = WARNING


[ENVOY]
; IP addess or FQDN of the Enphase Envoy
; Example: 192.168.0.123
; Example: envoy.local
address = envoy.local

; Time (seconds) between JSON requests
; default: 3600
; minimum: 900
fetch_production_historic_interval = 3600

; Device firmware
; for D5.x.x enter D5
; for D7.x.x enter D7
firmware = D7

; -- settings for D5.x.x firmware
; Requires the installer password
; can be generated with passwordCalc.py from this repository or https://github.com/sarnau/EnphaseEnergy
; you can test the credentials by opening http://IP_ADDR_OR_FQDN/installer/setup/home#overview
; with the username installer and the generated password
; Example: 12aB3C4d
password =

; -- settings for D7.x.x firmware
; Enphase Envoy serial number
; Example: 123456789012
serial = 122240001XXX
; Your Enphase Enlighten username
; Example: user@domain.tld
enlighten_user = XXXXXXXXXXXXXXX
; Your Enphase Enlighten password
; Example: topsecret123
enlighten_password = XXXXXXXXXXXXXXXX


[PV]
; Sum of the rated power (W) from all Microinverters connected to this Envoy
max = 11520
; position (connected input/output) of the PV inverter
; 0 = AC input 1 (select this if you don't have a MultiPlus or Quattro)
; 1 = AC output
; 2 = AC input 2
position = 0
; how many inverters are connected?
inverter_count = 30
; specify inverter type, e.g. M215, IQ6, IQ7A, IQ8A
inverter_type = IQ8HC


[DATA]
; Enables fetching of device status data (Microinverters, Q-Relays)
; 0 = Disabled
; 1 = Enabled
fetch_devices = 0
; Time (seconds) between requests for devices
; default: 3600
; minimum: 60
fetch_devices_interval = 3600
; How to publish data:
; 0 = publish only when data changed
; 1 = publish everytime
; default: 0
fetch_devices_publishing_type = 0

; Enables fetching of Microinverters production data
; 0 = Disabled
; 1 = Enabled
fetch_inverters = 0
; Time (seconds) between requests for inverters (the Envoy updates the data every 300 seconds, if not limiting the power)
; default: 300
; minimum: 60
fetch_inverters_interval = 300
; How to publish data:
; 0 = publish only when data changed
; 1 = publish everytime
; default: 0
fetch_inverters_publishing_type = 1

; Enable fetching of event data
; 0 = Disabled
; 1 = Enabled
fetch_events = 0
; Time (seconds) between requests for events
; default: 3600
; minimum: 900
fetch_events_interval = 3600
; How to publish data:
; 0 = publish only when data changed
; 1 = publish everytime
; default: 0
fetch_events_publishing_type = 0


[MQTT]
; Enables MQTT publishing
; 0 = Disabled
; 1 = Enabled
enabled = 0

; IP addess or FQDN from MQTT server
broker_address = IP_ADDR_OR_FQDN

; Port of the MQTT server
; default plaintext: 1883
; default TLS port: 8883
broker_port = 1883

; Time (seconds) between publishing data
; default: 5
; minimum value: 1
publish_interval = 5

; Enables TLS
; 0 = Disabled
; 1 = Enabled
;tls_enabled = 1

; Absolute path to the Certificate Authority certificate file that is to be treated as trusted by this client
;tls_path_to_ca = /data/keys/mosquitto.crt

; Disables verification of the server hostname in the server certificate
; 0 = Disabled
; 1 = Enabled
;tls_insecure = 1

; Username used for connection
;username = myuser

; Password used for connection
;password = mypassword

; Topic where the meters data as JSON string is published
topic_meters = enphase/envoy-s/meters

; Topic where the inverters data as JSON string is published
topic_inverters = enphase/envoy-s/inverters

; Topic where the devices data as JSON string is published
topic_devices = enphase/envoy-s/devices

; Topic where the events data as JSON string is published
topic_events = enphase/envoy-s/events
```

</details>


![IMG_0119](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/d1d7a226-ef36-4bf4-8ef0-6d81ed473ad7)

**Reply by mr-manuel** [reply, marked as answer]
In your config.ini you have set `position` to `0 = AC input 1`.

In your drawing you have connected the Enphase system to `AC output`.

Set the `position` to `1`, restart the driver and all should work.

PS: Please check your local Venus OS screen (remote console) before and after the change.

**Reply by Manuel-Calero** [reply]
Hello! I have made the change and reinstalled the service. Tomorrow when there is PV production I will check the loads again and tell you the result. Thank you for your support.

**Reply by Manuel-Calero** [reply]
I have made the change in the config.ini and it works. Thank you!

**Reply by mr-manuel** [reply]
Thanks you for your donation! I wish you a happy new year!

**Reply by Manuel-Calero** [reply]
The same to you!

**Comment by Manuel-Calero** [comment]
Hello again. I am thinking on the reason for the wrong data in my first post and I see in my very first Victron vrm picture:

![IMG_7508](https://github.com/mr-manuel/venus-os_dbus-enphase-envoy/assets/155190232/a8965970-cd79-4a13-8979-9d137939814b)

that the loads by fase for grid, are: 239W, -23W and -161W (the sum is 55W, not -2389 as indicated). So, I think the system has to sum 2444 (the production in the inverter) to -2389 (the grid label) = 55W to make the right calculation for Grid load.
